src/precice/impl/DataContext.cpp

```diff
@@ -10,8 +10,7 @@ DataContext::DataContext(mesh::PtrData data, mesh::PtrMesh mesh)
 {
   PRECICE_ASSERT(data);
   _providedData = data;
-  PRECICE_ASSERT(mesh);
-  _mesh = mesh;
+  _mesh         = mesh;
```

Comment:
This looks a bit hacky. `mesh` could now be a null pointer, right?

Reply:
yes, mesh=nullptr would fail the assertion so removed it to enable use with global data.

Besides we always have  `PRECICE_VALIDATE_DATA_NAME(meshName, dataName);` or `PRECICE_REQUIRE_DATA_WRITE/READ(meshName, dataName);` before construction of mesh data contexts which would already take care of asserting the mesh?

Reply:
Since it's hacky, will add it to the folow-up issue.